pkg/ingester/shipper.go

```diff
@@ -100,38 +102,28 @@ func NewShipper(
 // to the object bucket once.
 //
 // It is not concurrency-safe, however it is compactor-safe (running concurrently with compactor is ok).
-func (s *Shipper) Sync(ctx context.Context) (uploaded int, err error) {
-	meta, err := readShipperMetaFile(s.dir)
+func (s *Shipper) Sync(ctx context.Context) (shipped int, err error) {
+	shippedBlocks, err := readShippedBlocks(s.dir)
 	if err != nil {
 		// If we encounter any error, proceed with an empty meta file and overwrite it later.
 		// The meta file is only used to avoid unnecessary bucket.Exists call,
 		// which are properly handled by the system if their occur anyway.
-		if !os.IsNotExist(err) {
-			level.Warn(s.logger).Log("msg", "reading meta file failed, will override it", "err", err)
-		}
-		meta = &shipperMeta{Version: shipperMetaVersion1}
-	}
-
-	// Build a map of blocks we already uploaded.
-	hasUploaded := make(map[ulid.ULID]struct{}, len(meta.Uploaded))
-	for _, id := range meta.Uploaded {
-		hasUploaded[id] = struct{}{}
+		level.Warn(s.logger).Log("msg", "reading meta file failed, will override it", "err", err)
+		shippedBlocks = map[ulid.ULID]time.Time{}
 	}
 
-	// Reset the uploaded slice, so we can rebuild it only with blocks that still exist locally.
```

Comment:
[nit] I think this comment was helpful to clarify why we reset Shipped to an empty map.

Reply:
Added comment right above where we initialize the empty slice in case of error